pkg/pullsync/pullsync.go

```diff
@@ -238,7 +239,11 @@ func (s *Syncer) SyncInterval(ctx context.Context, peer swarm.Address, bin uint8
 		defer cancel()
 
 		if err := s.storage.Put(ctx, storage.ModePutSync, chunksToPut...); err != nil {
-			return topmost, fmt.Errorf("delivery put: %w", err)
+			if errors.Is(err, context.DeadlineExceeded) {
+				return topmost, fmt.Errorf("delivery put: %w", err)
+			} else {
```

Comment:
When `return` is used in the `if` block, wrapping the log message in the `else` block is unnecessary.